Build with Nix

Currently having trouble building in the Nix shell on Mac, due to a C compiler error:

```
# github.com/karalabe/usb
In file included from ../go/pkg/mod/github.com/karalabe/usb@v0.0.2/libs.go:50:
../go/pkg/mod/github.com/karalabe/usb@v0.0.2/libusb/libusb/os/darwin_usb.c:53:29: warning: macro 'ATOMIC_VAR_INIT' has been marked as deprecated [-Wdeprecated-pragma]
/nix/store/angs84gd7vrhjdpx47x04axa9m50x2y1-clang-wrapper-16.0.6/resource-root/include/stdatomic.h:54:41: note: macro marked 'deprecated' here
In file included from ../go/pkg/mod/github.com/karalabe/usb@v0.0.2/libs.go:50:
In file included from ../go/pkg/mod/github.com/karalabe/usb@v0.0.2/libusb/libusb/os/darwin_usb.c:72:
In file included from ../go/pkg/mod/github.com/karalabe/usb@v0.0.2/libusb/libusb/os/darwin_usb.h:26:
In file included from /nix/store/hmd0bywb7jfzya6d2amimdbj99gdfqwc-apple-framework-IOKit/Library/Frameworks/IOKit.framework/Headers/IOCFBundle.h:26:
In file included from /nix/store/cp5d54rghzzr9av9xxsl75977wx2w6g6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:60:
In file included from /nix/store/cp5d54rghzzr9av9xxsl75977wx2w6g6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFPropertyList.h:17:
In file included from /nix/store/cp5d54rghzzr9av9xxsl75977wx2w6g6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFStream.h:16:
/nix/store/cp5d54rghzzr9av9xxsl75977wx2w6g6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1189:83: error: expected ','
/nix/store/cp5d54rghzzr9av9xxsl75977wx2w6g6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1190:98: error: expected ','
/nix/store/cp5d54rghzzr9av9xxsl75977wx2w6g6-apple-framework-CoreFoundation-11.0.0/Library/Frameworks/CoreFoundation.framework/Headers/CFURL.h:1200:85: error: expected ','
```

Works fine outside of the Nix shell, using
```
$ go version
go version go1.20.7 darwin/amd64
$ cc --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: x86_64-apple-darwin22.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

--- a/core/gen_genesis.go
+++ b/core/gen_genesis.go
@@ -31,6 +31,7 @@ func (g Genesis) MarshalJSON() ([]byte, error) {
 		GasUsed    math.HexOrDecimal64                         `json:"gasUsed"`
 		ParentHash common.Hash                                 `json:"parentHash"`
 		BaseFee    *math.HexOrDecimal256                       `json:"baseFeePerGas"`
+		StateHash  *common.Hash                                `json:"stateHash,omitempty"`
 	}
 	var enc Genesis
 	enc.Config = g.Config
@@ -51,6 +52,7 @@ func (g Genesis) MarshalJSON() ([]byte, error) {
 	enc.GasUsed = math.HexOrDecimal64(g.GasUsed)
 	enc.ParentHash = g.ParentHash
 	enc.BaseFee = (*math.HexOrDecimal256)(g.BaseFee)
+	enc.StateHash = g.StateHash
 	return json.Marshal(&enc)
 }
 
@@ -70,6 +72,7 @@ func (g *Genesis) UnmarshalJSON(input []byte) error {
 		GasUsed    *math.HexOrDecimal64                        `json:"gasUsed"`
 		ParentHash *common.Hash                                `json:"parentHash"`
 		BaseFee    *math.HexOrDecimal256                       `json:"baseFeePerGas"`
+		StateHash  *common.Hash                                `json:"stateHash,omitempty"`
 	}
 	var dec Genesis
 	if err := json.Unmarshal(input, &dec); err != nil {
@@ -119,6 +122,9 @@ func (g *Genesis) UnmarshalJSON(input []byte) error {
 	}
 	if dec.BaseFee != nil {
 		g.BaseFee = (*big.Int)(dec.BaseFee)
+	}
+	if dec.StateHash != nil {
+		g.StateHash = dec.StateHash
 	}
 	return nil
 }

--- a/flake.nix
+++ b/flake.nix
@@ -63,6 +63,10 @@
 
             # geth node
             go-ethereum
+          ] ++ lib.optionals stdenv.isDarwin [
+            darwin.libobjc
+            darwin.IOKit
+            darwin.apple_sdk.frameworks.CoreFoundation
           ];
         };
       });

